lib/testsuite/index.js

```diff
@@ -838,7 +845,36 @@ class TestSuite {
 
     return runnable.run(this.commandQueue);
   }
-  
+
+  takeSnapshot(node) {
+    const commands = this.reporter.currentSection.commands || [];
+    const currentCommand = commands[commands.length-1];
+
+    if (this.settings.trace.enabled && node.isTraceable) {
+      const snapShotPath = Snapshots.getFileName({
+        testSuite: this.api.currentTest.module,
+        commandName: node.fullName,
+        traceSettings: this.settings.trace,
+        output_folder: this.settings.output_folder
+      });
+
+      this.__snapShot = new Promise((resolve, reject) => {
+        this.api.saveSnapshot(snapShotPath, (result => {
```

Comment:
saveSnapshot can be just a util method, it doesn't have to be a command.

Reply:
resolving based on discussion